### PR TITLE
support multi-schema change for add columns

### LINKED ISSUE #4: Support multi-schema change for `add columns`

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close https://github.com/tangenta/tidb/issues/4

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: ddl/multi_schema_change.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+			proxyJob := cloneFromSubJob(job, sub)
+			ver, err = w.runDDLJob(d, t, proxyJob)
+			mergeBackToSubJob(proxyJob, sub)
+			handleRevertibleException(job, sub.State, i)
+			return ver, err
+		}
+		// All the sub-jobs are non-revertible.
+		job.MultiSchemaInfo.Revertible = false
+		// Step the sub-jobs to the non-revertible states all at once.
+		for _, sub := range job.MultiSchemaInfo.SubJobs {
```
Comment (Defined2014) [inline ddl/multi_schema_change.go:67]: why need this here? Maybe L74-L83 is enough.

File: ddl/column.go [hunk 2]
```diff
@@ -402,7 +373,7 @@ func onAddColumns(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, err error
 				}
 			}
 			tblInfo.Columns = append(tblInfo.Columns, newCols[i])
-			adjustColumnInfoInAddColumn(tblInfo, offsets[i])
+			tblInfo.MoveColumnInfo(len(tblInfo.Columns)-1, offsets[i])
```
Comment (Defined2014) [inline ddl/column.go:376]: Maybe `onAddColumns` will unreachable after this PR.

Comment (tangenta) [reply ddl/column.go:376]: Nope, the DDL like `alter table t add column a int, b int` will still use `onAddColumns`.

Comment (Defined2014) [reply ddl/column.go:376]: It's an illegal stmt in mysql. Shoud we keep this?

File: ddl/multi_schema_change.go (hunk 1 above)
Comment (tangenta) [reply ddl/multi_schema_change.go:67]: This is used to step all the schema object to a "visible" state. Note that there is no "return" in the loop.

Take `add column a, add column b` as an example, this is used to change `a` and `b` to public in a single transaction.

File: ddl/column.go (hunk 2 above)
Comment (Defined2014) [reply ddl/column.go:376]: Seems we support this stmt in parser for a long time. Just keep it right now.

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**